### PR TITLE
fix reusing shas for deployment events

### PR DESCRIPTION
**Pull request by xh3b4sd**
Towards https://github.com/giantswarm/giantswarm/issues/1616.

### CODE REVIEW

File: service/eventer/github/client.go [hunk 1]
```diff
@@ -44,35 +44,27 @@ func (e *Eventer) request(req *http.Request) (*http.Response, error) {
...
 		// If there are any statuses apart from pending, we consider the
 		// deployment finished, and do not act on it.
-		isPending := true
-		for _, status := range deployment.Statuses {
-			if status.State != pendingState {
-				isPending = false
-				break
-			}
+		if len(deployment.Statuses) != 0 {
+			continue
```
Comment (xh3b4sd) [inline service/eventer/github/client.go:56]: Now we filter everything that does not have any status. In the next steps the eventer will set a pending state to the deployment event and the operator will set a success or failure state.

File: service/eventer/github/client.go [hunk 2]
```diff
@@ -44,35 +44,27 @@ func (e *Eventer) request(req *http.Request) (*http.Response, error) {
...
 
 	return matches
 }
 
 // fetchNewDeploymentEvents fetches any new GitHub Deployment Events for the
 // given project.
-func (e *Eventer) fetchNewDeploymentEvents(project, environment string, etagMap map[string]string, filterFinished bool) ([]deployment, error) {
-	e.logger.Log("debug", "fetching deployments", "project", project)
-
+func (e *Eventer) fetchNewDeploymentEvents(project, environment string, etagMap map[string]string) ([]deployment, error) {
```
Comment (xh3b4sd) [inline service/eventer/github/client.go:67]: Cleaning up the interface since we always use `FetchLatest`.

File: service/eventer/github/client.go [hunk 3]
```diff
@@ -112,8 +104,7 @@ func (e *Eventer) fetchNewDeploymentEvents(project, environment string, etagMap
 	etagMap[project] = resp.Header.Get(etagHeader)
 
 	if resp.StatusCode == http.StatusNotModified {
-		e.logger.Log("debug", "no new deployment events, continuing", "project", project)
-		return nil, nil
+		return nil, microerror.Mask(notFoundError)
```
Comment (xh3b4sd) [inline service/eventer/github/client.go:107]: The caller decides what to do when there are no deployment events.

File: service/eventer/github/client.go [hunk 4]
```diff
@@ -138,14 +129,10 @@ func (e *Eventer) fetchNewDeploymentEvents(project, environment string, etagMap
...
 
-	if filterFinished {
-		e.logger.Log("debug", "filtering finished deployment events", "project", project)
-		deployments = e.filterDeploymentsByStatus(deployments)
-	}
-
-	if len(deployments) > 0 {
-		e.logger.Log("debug", fmt.Sprintf("found %d new deployment events", len(deployments)), "project", project)
+	if len(deployments) == 0 {
+		return nil, microerror.Mask(notFoundError)
```
Comment (xh3b4sd) [inline service/eventer/github/client.go:135]: The caller decides what to do when there are no deployment events.

File: service/eventer/github/github.go [hunk 5]
```diff
@@ -104,21 +104,21 @@ func (e *Eventer) FetchContinuously(projects []string, environment string) (<-ch
...
 	go func() {
 		for {
 			select {
 			case <-ticker.C:
 				for _, p := range projects {
-					deployments, err := e.fetchNewDeploymentEvents(p, environment, e.etagMap, filterFinished)
-					if err != nil {
+					d, err := e.FetchLatest(p, environment)
+					if IsNotFound(err) {
+						continue
```
Comment (xh3b4sd) [inline service/eventer/github/github.go:115]: Here we are good in case there are no new events.

File: service/eventer/github/github.go [hunk 6]
```diff
@@ -104,21 +104,21 @@ func (e *Eventer) FetchContinuously(projects []string, environment string) (<-ch
...
+					} else if err != nil {
 						e.logger.Log("error", "could not fetch deployment events", "message", err.Error())
+						continue
 					}
 
-					for _, d := range deployments {
-						deploymentEventChannel <- d.DeploymentEvent(p)
-					}
+					deploymentEventChannel <- d
```
Comment (xh3b4sd) [inline service/eventer/github/github.go:121]: Now we only process the latest one. Before we overwrote events.

**Review by JosephSalisbury — APPROVED** [review]

**Pull request merged**